src/main/java/org/broadinstitute/hellbender/tools/dataflow/pipelines/ReadsPreprocessingPipeline.java

```diff
@@ -96,21 +101,42 @@ protected void setupPipeline( Pipeline pipeline ) {
         // Load the Variants and the Reference and join them to reads.
         final VariantsDataflowSource variantsDataflowSource = new VariantsDataflowSource(baseRecalibrationKnownVariants, pipeline);
 
-        Map<String, String> referenceNameToIdTable = RefAPISource.buildReferenceNameToIdTable(pipeline.getOptions(), referenceName);
+        String referenceName = RefAPISource.getReferenceSetID(referenceURL);
+        final RefAPISource ref = RefAPISource.getInstance();
+        Map<String, Reference> referenceMap = ref.getReferenceNameToReferenceTable(pipeline.getOptions(), referenceName);
+        final Map<String, String> referenceNameToIdTable = ref.getReferenceNameToIdTableFromMap(referenceMap);
+
         // Use the BQSR_REFERENCE_WINDOW_FUNCTION so that the reference bases required by BQSR for each read are fetched
         RefAPIMetadata refAPIMetadata = new RefAPIMetadata(referenceName, referenceNameToIdTable, BaseRecalibratorDataflow.BQSR_REFERENCE_WINDOW_FUNCTION);
 
         final PCollection<KV<GATKRead, ReadContextData>> readsWithContext = AddContextDataToRead.add(markedReads, refAPIMetadata, variantsDataflowSource);
 
         // Apply BQSR.
-        final PCollection<RecalibrationTables> recalibrationReports = readsWithContext.apply(new BaseRecalibratorStub(headerSingleton));
-        final PCollectionView<RecalibrationTables> mergedRecalibrationReport = recalibrationReports.apply(View.<RecalibrationTables>asSingleton());
+        // default arguments are best practice.
+        BaseRecalibrationArgumentCollection BRAC = new BaseRecalibrationArgumentCollection();
+        final SAMSequenceDictionary readsDictionary = readsHeader.getSequenceDictionary();
+        final SAMSequenceDictionary refDictionary = ref.getReferenceSequenceDictionaryFromMap(referenceMap, readsDictionary);
+        checkSequenceDictionaries(refDictionary, readsDictionary);
+        PCollectionView<SAMSequenceDictionary> refDictionaryView = pipeline.apply(Create.of(refDictionary)).setName("refDictionary").apply(View.asSingleton());
+        final PCollection<BaseRecalOutput> recalibrationReports = readsWithContext.apply(new BaseRecalibratorTransform(headerSingleton, refDictionaryView, BRAC));
+
+        // pretend to apply BQSR
+        //final PCollection<BaseRecalOutput> recalibrationReports = readsWithContext.apply(new BaseRecalibratorStub(headerSingleton));
```

Comment:
Remove commented-out call to stub, and delete the `BaseRecalibratorStub` itself.